t/encoding.t

```diff
@@ -0,0 +1,19 @@
+use strict;
+use warnings;
+
+use Test::More;
+
+use HTTP::Daemon ();
+
+{
+    no warnings 'redefine';
+    local *IO::Socket::IP::sockhost
+        = sub { return q{fe80::250:54ff:fe00:f01%ens3} };
+
+    my $d = HTTP::Daemon->new;
+    is($d->sockhost, q{fe80::250:54ff:fe00:f01%ens3}, 'we overrode sockhost');
+
+    like($d->url, qr{[fe80::250:54ff:fe00:f01%25ens3]}, '% is encoded in host');
```

Comment:
I think we need to escape `[`.
```
like($d->url, qr{\Q[fe80::250:54ff:fe00:f01%25ens3]\E}, '% is encoded in host');
```

Reply:
Oh, good catch! Thanks. I've updated the test and force-pushed.